tools/create-genesis/src/genesis.rs

```diff
@@ -0,0 +1,235 @@
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Account {
+    pub nonce: U256,
+    pub code: String,
+    pub storage: BTreeMap<String, String>,
+    pub value: U256,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct Genesis {
+    pub timestamp: u64,
+    pub prevhash: String,
+    pub alloc: BTreeMap<String, Account>,
+}
+
+impl Default for Genesis {
+    fn default() -> Self {
+        Genesis {
+            timestamp: 0,
+            prevhash: String::default(),
+            alloc: BTreeMap::new(),
+        }
+    }
+}
+
+impl<'a> GenesisCreator<'a> {
+    pub fn new(
+        contract_dir: &'a str,
+        contract_docs_dir: &'a str,
+        params_path: &'a str,
+        genesis_path: &'a str,
+        timestamp: u64,
+        init_token: &'a str,
+        prevhash: &'a str,
+    ) -> Self {
+        let params = InitData::load_contract_args(params_path);
+        let contracts_list = contract_dir.to_owned() + "/contracts.yml";
+        let constracts = ContractsData::load_contract_list(&contracts_list);
+
+        GenesisCreator {
+            contract_dir,
+            contract_docs_dir,
+            genesis_path,
+            timestamp,
+            init_token,
+            prevhash,
+            contract_args: params,
+            contract_list: constracts,
+            accounts: BTreeMap::new(),
+        }
+    }
+
+    pub fn create(&mut self) {
+        // 1. Check compile exit or not
+        if !Solc::compiler_version() {
+            panic!("solc compiler not exit");
+        }
+        // 2. Init normal contracts
+        self.init_normal_contracts();
+        // 3. Init permission contracts
+        self.init_permission_contracts();
+        // 4. Save super admin
+        let super_admin = self.contract_args.contracts.admin.admin.clone();
+        self.set_account_value(
+            &super_admin,
+            U256::from_str(clean_0x(&self.init_token)).unwrap(),
+        );
+        // 5. Save genesis to file
+        self.save_to_file();
+        println!("Create genesis successfully !");
+    }
+
+    pub fn init_normal_contracts(&mut self) {
+        for (contract_name, contract_info) in self.contract_list.normal_contracts.list().iter() {
+            let address = &contract_info.address;
+            let sol_path = self.contract_dir.to_owned() + "/src/" + &contract_info.file;
+            let abi_path =
+                self.contract_dir.to_owned() + "/interaction/abi/" + contract_name + ".abi";
+            let data = Solc::get_contracts_data(sol_path, contract_name);
+
+            let input_data = string_2_bytes(data["bin"].clone());
+            self.write_docs(contract_name, data);
+
+            let abi_file = File::open(abi_path).expect("failed to open abi file.");
+            let contract = Contract::load(abi_file).unwrap();
+
+            if let Some(constructor) = contract.constructor() {
+                let mut params = Vec::new();
+                match *contract_name {
```

Comment:
These codes are ugly...
Could we refactor all the same logic?

Reply:
It can be done.  Refactor works should be in another pr later.

Reply:
Later means never, for most situations(especially for cita).

THIS IS NOT ANOTHER PR THING.IT'S ABOUT THE CODE THAT WILL BE MERGED.

Why another pr? Why refactor works should be in another pr?  I think refactoring the exist codes should be in another pr instead of the codes of this pr. Is it not already a refactoring for create_genesis?
Is the quality not the review opition for a pr?

Sorry, I don't accept the reason. and i cannot approve it.

Reply:
ok, wait a while